Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Analytics.java

```diff
@@ -0,0 +1,104 @@
+package org.dolphinemu.dolphinemu.utils;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.IntentFilter;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.support.v4.content.LocalBroadcastManager;
+
+import com.android.volley.Request;
+import com.android.volley.toolbox.StringRequest;
+
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.features.settings.model.Settings;
+import org.dolphinemu.dolphinemu.features.settings.utils.SettingsFile;
+import org.dolphinemu.dolphinemu.services.DirectoryInitializationService;
+
+public class Analytics
+{
+	private static DirectoryStateReceiver directoryStateReceiver;
+	private static String analyticsAsked = Settings.SECTION_ANALYTICS + "_" + SettingsFile.KEY_ANALYTICS_PERMISSION_ASKED;
+	private static final String DEVICE_MANUFACTURER = "DEVICE_MANUFACTURER";
+	private static final String DEVICE_OS = "DEVICE_OS";
+	private static final String DEVICE_MODEL = "DEVICE_MODEL";
+
+	public static void checkAnalyticsInit(Context context)
+	{
+		SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
+		if (!preferences.getBoolean(analyticsAsked, false))
+		{
+			if (!DirectoryInitializationService.areDolphinDirectoriesReady())
+			{
+				// Wait for directories to get initialized
+				IntentFilter statusIntentFilter = new IntentFilter(
+						DirectoryInitializationService.BROADCAST_ACTION);
+
+				directoryStateReceiver = new DirectoryStateReceiver(directoryInitializationState ->
+				{
+					if (directoryInitializationState == DirectoryInitializationService.DirectoryInitializationState.DOLPHIN_DIRECTORIES_INITIALIZED)
+					{
+						LocalBroadcastManager.getInstance(context).unregisterReceiver(directoryStateReceiver);
+						directoryStateReceiver = null;
+						showMessage(context, preferences);
+					}
+				});
+				// Registers the DirectoryStateReceiver and its intent filters
+				LocalBroadcastManager.getInstance(context).registerReceiver(
+						directoryStateReceiver,
+						statusIntentFilter);
+			}
+			else
+				showMessage(context, preferences);
+		}
+	}
+
+	private static void showMessage(Context context, SharedPreferences preferences)
+	{
+		// We asked, set to true regardless of answer
+		SharedPreferences.Editor sPrefsEditor = preferences.edit();
+		sPrefsEditor.putBoolean(analyticsAsked, true);
+		sPrefsEditor.apply();
+
+		new AlertDialog.Builder(context)
+				.setMessage(context.getString(R.string.analytics_enable))
```

Comment:
The prompt on initial launch needs a better description. Right now we don't really tell the user what usage statistics reporting does.